Tests/kaas/k8s-default-storage-class/k8s-default-storage-class-check.py

```diff
@@ -0,0 +1,311 @@
+#!/usr/bin/env python3
+
+"""PersistentVolumeClaims checker
+
+Return codes:
+0:    Default StorageClass is available, and setup to SCS standard
+1:    Not able to connect to k8s api
+
+31:   Default storage class has no provisioner
+32:   None or more then one default Storage Class is defined
+
+41:   Not able to bind PersitantVolume to PersitantVolumeClaim
+42:   ReadWriteOnce is not a supported access mode
+
+All return codes between (and including) 1-19 as well as all return codes ending on 9
+can be seen as failures.
+
+Check given cloud for conformance with SCS standard regarding
+Default StorageClass and PersistentVolumeClaims, to be found under /Standards/scs-0211-v1-kaas-default-storage-class.md
+
+"""
+
+import getopt
+import sys
+import time
+import json
+import logging
+
+from kubernetes import client, config
+from helper import gen_sonobuoy_result_file
+
+import logging.config
+
+logger = logging.getLogger("k8s-default-storage-class-check")
+
+
+def setup_k8s_client(kubeconfigfile=None):
+
+    if kubeconfigfile:
+        logger.debug(f"using kubeconfig file '{kubeconfigfile}'")
+        config.load_kube_config(kubeconfigfile)
+    else:
+        logger.debug(" useing system kubeconfig")
+        config.load_kube_config()
+
+    k8s_api_client = client.CoreV1Api()
+    k8s_storage_client = client.StorageV1Api()
+
+    return (
+        k8s_api_client,
+        k8s_storage_client,
+    )
+
+
+def print_usage(file=sys.stderr):
+    """Help output"""
+    print(
+        """Usage: k8s_storageclass_check.py [options]
+This tool checks the requested k8s default storage class according to the SCS Standard 0211 "kaas-default-storage-class".
+Options:
+ [-k/--kubeconfig PATH_TO_KUBECONFIG] sets kubeconfig file to access kubernetes api
+ [-d/--debug] enables DEBUG logging channel
+""",
+        end="",
+        file=file,
+    )
+
+
+def initialize_logging():
+    logging.basicConfig(format="%(levelname)s: %(message)s", level=logging.INFO)
+
+
+class SCSTestException(Exception):
+    """Raised when an Specific test did not pass"""
+
+    def __init__(self, *args, return_code: int):
+        self.return_code = return_code
+
+
+def check_default_storageclass(k8s_client_storage):
+
+    api_response = k8s_client_storage.list_storage_class(_preload_content=False)
+    storageclasses = api_response.read().decode("utf-8")
+    storageclasses_dict = json.loads(storageclasses)
+
+    ndefault_class = 0
+
+    for item in storageclasses_dict["items"]:
+        storage_class_name = item["metadata"]["name"]
+        annotations = item["metadata"]["annotations"]
+
+        if annotations["storageclass.kubernetes.io/is-default-class"] == "true":
+            ndefault_class += 1
+            default_storage_class = storage_class_name
+            provisioner = item["provisioner"]
+
+    if provisioner == "kubernetes.io/no-provisioner":
+        raise SCSTestException(
```

Comment:
I do not favor to exit script at first error. This makes debugging and error fixing really hard for CSPs. I prefer to collect all errors and print an error report as result of the script.